psl/psl-core/src/validate/datasource_loader.rs

```diff
@@ -49,85 +53,67 @@ fn lift_datasource(
     diagnostics: &mut Diagnostics,
     connectors: crate::ConnectorRegistry,
 ) -> Option<Datasource> {
-    let source_name = &ast_source.name.name;
-    let mut args: HashMap<_, _> = ast_source
+    let source_name = ast_source.name.name.as_str();
+    let mut args: HashMap<_, (_, &Expression)> = ast_source
         .properties
         .iter()
-        .map(|arg| (arg.name.name.as_str(), (arg.span, &arg.value)))
-        .collect();
+        .map(|arg| match &arg.value {
+            Some(expr) => Some((arg.name.name.as_str(), (arg.span, expr))),
+            None => {
+                diagnostics.push_error(DatamodelError::new_config_property_missing_value_error(
+                    &arg.name.name,
+                    source_name,
+                    "datasource",
+                    ast_source.span,
+                ));
+                None
+            }
+        })
+        .collect::<Option<HashMap<_, (_, _)>>>()?;
 
-    let (_, provider_arg) = match args.remove("provider") {
-        Some(provider) => provider,
-        None => {
-            diagnostics.push_error(DatamodelError::new_source_argument_not_found_error(
-                "provider",
-                &ast_source.name.name,
-                ast_source.span,
-            ));
-            return None;
-        }
-    };
+    // (arg.name.name.as_str(), (arg.span, &arg.value))
```

Comment:
Don't leave commented lines like that in the codebase. I can point you to commented code such as this from years ago, with some message "TODO: figure out later". We have Git for version history!